runtime/jcl/common/jclcinit.c

```diff
@@ -66,87 +67,85 @@ jint computeFullVersionString(J9JavaVM* vm)
 {
 	VMI_ACCESS_FROM_JAVAVM((JavaVM*)vm);
 	PORT_ACCESS_FROM_JAVAVM(vm);
+	OMRPORT_ACCESS_FROM_J9PORT(vm->portLibrary);
 	const char *osarch;
 	const char *osname;
 	const char *vmVersion = NULL;
 	const char *gcVersion = NULL;
-	char fullversion[512];
 	char vminfo[512];
+        char fullversion[512];
+        int buflen = 0;
+        int vmlen = 0;  
 #ifdef J9VM_INTERP_NATIVE_SUPPORT
 	J9JITConfig *jitConfig;
 #endif
 #if defined(J9VM_INTERP_NATIVE_SUPPORT)
 	UDATA jitEnabled = 0;
 	UDATA aotEnabled = 0;
 #endif
-
-	strcpy(fullversion, "JRE ");
-	strcpy(vminfo, "JRE ");
+        buflen = omrstr_printf(fullversion, sizeof(fullversion), "JRE ");
+        vmlen = omrstr_printf(vminfo, sizeof(vminfo), "JRE ");
 	switch(J2SE_VERSION(vm) & J2SE_VERSION_MASK) {
 	case J2SE_17:
 		if ((J2SE_VERSION(vm) & J2SE_RELEASE_MASK) == J2SE_17) {
-			strcat(fullversion, "1.7.0 ");
-			strcat(vminfo, "1.7.0 ");
+		        buflen += omrstr_printf(fullversion+buflen, sizeof(fullversion), "1.7.0 ");	
```

Comment:
Please remove the whitespace which was added at the end of most of the omrstr_printf() lines.

Reply:
The correct length is sizeof(fullversion)-buflen. Similarly for vminfo.